src/composable/conditionalorder.ts

```diff
@@ -0,0 +1,253 @@
+import { BigNumber, ethers, utils } from 'ethers'
+import { ContextFactory } from './multiplexer'
+import { keccak256 } from 'ethers/lib/utils'
+
+// Define the ABI tuple for the TWAPData struct
+export const CONDITIONAL_ORDER_PARAMS_ABI = ['tuple(address handler, bytes32 salt, bytes staticInput)']
+
+export type ConditionalOrderParams = {
+  readonly handler: string
+  readonly salt: string
+  readonly staticInput: string
+}
+
+/**
+ * An abstract base class from which all conditional orders should inherit.
+ *
+ * This class provides some basic functionality to help with handling conditional orders,
+ * such as:
+ * - Validating the conditional order
+ * - Creating a human-readable string representation of the conditional order
+ * - Serializing the conditional order for use with the `IConditionalOrder` struct
+ * - Getting any dependencies for the conditional order
+ * - Getting the off-chain input for the conditional order
+ *
+ * **NOTE**: Instances of conditional orders have an `id` property that is a `keccak256` hash of
+ *           the serialized conditional order.
+ */
+export abstract class BaseConditionalOrder<T, P> {
+  public readonly handler: string
```

Comment:
Just a good to have:
There are list of utils we collected during TWAP integration:
https://github.com/cowprotocol/cowswap/tree/develop/src/modules/twap/utils

Some of them I've found in this PR some of them not. If you see something helpful - please feel free to add it to the SDK.
It's not necessary, just good to have.

Reply:
As front-end is a consumer downstream, it would be handy to know what you'd like included 😃

Anything that you feel particularly passionate about, or you think would benefit consumers of `cow-sdk`?

Reply:
I've gone through and had a look in the utils that were created for TWAP. Most of these exist in the `TWAP` class, with the exception of the discrete order generator that I've mentioned (ie. given a parent conditional order, generate a list of all discrete orders).

Implementing the discrete order generators is tracked in #144